IntelliTect.TestTools.Selenate/IntelliTect.TestTools.SelenateExtensions/WebElementExtensions.cs

```diff
@@ -58,34 +59,52 @@ public static async Task FillInWithAndTabWhenReady(this IWebElement element, str
         /// or throws after a certain amount of time
         /// </summary>
         /// <param name="value">Value to send to the element</param>
-        public static async Task FillInWithWhenReady(this IWebElement element, string value, int secondstoTry = 5)
+        public static void FillInWithWhenReady(this IWebElement element, IWebDriver driver, string value, int secondstoTry = 5)
         {
-            var count = 0;
-            while (element.GetAttribute("value") != value && count < 5)
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(secondstoTry));
+            wait.IgnoreExceptionTypes(
+                typeof(ElementNotVisibleException),
+                typeof(ElementNotInteractableException),
+                typeof(StaleElementReferenceException),
+                typeof(InvalidElementStateException));
+
+            for ( int i = 0; i < 5; i++ )
             {
-                await Wait.Until<NoSuchElementException, 
-                    ElementNotVisibleException,
-                    StaleElementReferenceException>(
-                    () => element.Clear(), TimeSpan.FromSeconds(secondstoTry));
-                await Wait.Until<NoSuchElementException, 
-                    ElementNotVisibleException, 
-                    ElementNotInteractableException,
-                    StaleElementReferenceException>(
-                    () => element.SendKeys(value), TimeSpan.FromSeconds(secondstoTry));
-                count++;
+                wait.Until( c =>
+                {
+                    element.Clear();
+                    return true;
```

Comment:
Why these return values? `WebDriver.Until` appears to have an `Action` overload so you should not need to have a return value unless you intend to use it somewhere.

Reply:
The only one I see is for a Func<IWebDriver, bool> argument?

This is the class that WebDriverWait implements:
https://github.com/SeleniumHQ/selenium/blob/master/dotnet/src/webdriver/Support/DefaultWait%7BT%7D.cs